### PR TITLE
Generate protos

### LINKED ISSUE #1: Consume opamp-spec as a submodule and use protos from it
Depends on https://github.com/open-telemetry/opamp-go/issues/133

This is similar to what we do with OTLP protos in https://github.com/open-telemetry/opentelemetry-proto-go

We need a makefile to re-generate the files every time protos are updated. Generated files should be stored in this repo.

Ideally also a github action that verifies that protos match the generated file.

### PR DESCRIPTION
**Pull request by jlegoff**
This is a first attempt at setting up the build files and generating the protos.

Since the generated protos are part of the repo, the generation itself is not part of the build pipeline and is set up as an standalone project in the `generateProtos` directory. Regenerating them can be done manually as explained in the README.

While the issue mentions a makefile for proto generation, I took inspiration from [opentelemetry-proto-java](https://github.com/open-telemetry/opentelemetry-proto-java) and used a gradle script instead. I can change that if needed.

Note also that the generated code is not published as a library. I suggest doing this in a separate PR.

Resolves #1

### CODE REVIEW

File: .gitmodules
Comment (jack-berg) [inline .gitmodules:4]: Not familiar with `.gitmodules` files, but the info in here points to the wrong repository. We don't have these files in `opentelemetry-java` or `opentelemetry-java-instrumentation`. Care to summarize their purpose?

File: settings.gradle.kts
Comment (jack-berg) [inline settings.gradle.kts]: Current latest version of this [plugin](https://github.com/google/protobuf-gradle-plugin) is `0.8.19`.

File: proto/build.gradle.kts
Comment (jack-berg) [inline proto/build.gradle.kts:23]: What's the advantage of doing the submodule thing referencing the `opamp-spec` repo versus defining a task to download the protos from the `opamp-spec` into the build dir?

File: proto/build.gradle.kts
Comment (jack-berg) [inline proto/build.gradle.kts]: These should come from `otel.java-conventions`.

File: .gitmodules
Comment (jlegoff) [reply .gitmodules:4]: the opamp-spec repo is used to get the proto definitions, as suggested in the issue. Here I set up a submodule and pinned it to the latest release.

File: proto/build.gradle.kts
Comment (jlegoff) [reply proto/build.gradle.kts:23]: this is a good question. This is what is suggested in the repo, but I can change it do behave like opentelemetry-java-protos. Perhaps @tigrannajaryan has an opinion here.

File: proto/build.gradle.kts
Comment (jlegoff) [reply proto/build.gradle.kts]: thanks, I removed this section

File: settings.gradle.kts
Comment (jlegoff) [reply settings.gradle.kts]: updated, thanks

**Comment by tigrannajaryan** [comment]
Hey folks, great to see progress on this but I want to make sure we set the expectations right. Unless we find the second long-term maintainer we will not be able to move forward with this repo. It is not healthy to have a repo with only one long-term maintainer.

@jack-berg since you are reviewing this are you interested in being the second maintainer?

File: proto/build.gradle.kts
Comment (AvadheshKaria) [reply proto/build.gradle.kts:23]: I like git submodule, unless there are any other advantages w.r.t. task to download the protos.

**Review by AvadheshKaria — APPROVED** [review]

**Review by AvadheshKaria — APPROVED** [review]

File: proto/build.gradle.kts
Comment (tigrannajaryan) [reply proto/build.gradle.kts:23]: > this is a good question. This is what is suggested in the repo, but I can change it do behave like opentelemetry-java-protos. Perhaps @tigrannajaryan has an opinion here.

@jlegoff I used the submodule in opamp-go since it was the simplest way I knew to bring it as a dependency pinned to a specific commit. If using a gradle download task has any advantages feel free to do that instead. What do you gain by doing that?

Comment (jlegoff) [reply proto/build.gradle.kts:23]: Honestly, I don't know what we gain by using a download tasks instead of a submodule. Seems like a matter of taste. @jack-berg do you have a strong opinion against using submodules?

Comment (jack-berg) [reply proto/build.gradle.kts:23]: Nope, no strong opinion. Agree it feels like a matter of taste.

**Comment by jack-berg** [comment]
Hey @tigrannajaryan - I'm happy to act in an approver capacity for this repository, but not able to commit to be a maintainer.

If we need a second maintainer (which I think is a very reasonable requirement), someone else will have to volunteer.

**Review by jack-berg — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Thanks @jack-berg, your reviews/approvals are appreciated.

We now also have @AvadheshKaria who I believe will work on becoming the long term maintainer of this repo.

**Comment by tigrannajaryan** [comment]
@jlegoff we now have approvals. I can merge this if you don't plan any other changes to this PR.

**Comment by jlegoff** [comment]
@tigrannajaryan great, let's merge it and I'll work on the next issue later this week

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**
